fdbserver/SimKmsConnector.actor.cpp

```diff
@@ -66,90 +71,160 @@ struct SimKmsConnectorContext {
+		dbgKIdTrace.get().setMaxEventLength(100000);
+		dbgKIdTrace.get().detail("DbgId", req.debugId.get());
+	}
+
+	// Lookup corresponding EncryptKeyCtx for input keyId
+	for (const auto& item : req.encryptKeyIds) {
+		const auto& itr = ctx->simEncryptKeyStore.find(item.first);
+		if (itr != ctx->simEncryptKeyStore.end()) {
+			rep.cipherKeyDetails.emplace_back(
+			    item.second, itr->first, StringRef(rep.arena, itr->second.get()->key), rep.arena);
+
+			if (dbgKIdTrace.present()) {
+				// {encryptDomainId, baseCipherId} forms a unique tuple across encryption domains
+				dbgKIdTrace.get().detail(
+				    getEncryptDbgTraceKey(ENCRYPT_DBG_TRACE_RESULT_PREFIX, item.second, itr->first), "");
+			}
+		} else {
+			success = false;
+			break;
+		}
+	}
+
+	wait(delayJittered(1.0)); // simulate network delay
+
+	success ? req.reply.send(rep) : req.reply.sendError(encrypt_key_not_found());
+
+	return Void();
+}
+
+ACTOR Future<Void> ekLookupByDomainIds(Reference<SimKmsConnectorContext> ctx,
+                                       KmsConnectorInterface interf,
+                                       KmsConnLookupEKsByDomainIdsReq req) {
+	state KmsConnLookupEKsByDomainIdsRep rep;
+	state bool success = true;
+	state Optional<TraceEvent> dbgDIdTrace =
+	    req.debugId.present() ? TraceEvent("SimKmsGetsByDomIds", interf.id()) : Optional<TraceEvent>();
+
+	if (dbgDIdTrace.present()) {
+		dbgDIdTrace.get().detail("DbgId", req.debugId.get());
+	}
+
+	// Map encryptionDomainId to corresponding EncryptKeyCtx element using a modulo operation. This
+	// would mean multiple domains gets mapped to the same encryption key which is fine, the
+	// EncryptKeyStore guarantees that keyId -> plaintext encryptKey mapping is idempotent.
+	for (EncryptCipherDomainId domainId : req.encryptDomainIds) {
+		EncryptCipherBaseKeyId keyId = 1 + abs(domainId) % SERVER_KNOBS->SIM_KMS_MAX_KEYS;
+		const auto& itr = ctx->simEncryptKeyStore.find(keyId);
+		if (itr != ctx->simEncryptKeyStore.end()) {
+			rep.cipherKeyDetails.emplace_back(domainId, keyId, StringRef(itr->second.get()->key), rep.arena);
+			if (dbgDIdTrace.present()) {
+				// {encryptId, baseCipherId} forms a unique tuple across encryption domains
+				dbgDIdTrace.get().detail(getEncryptDbgTraceKey(ENCRYPT_DBG_TRACE_RESULT_PREFIX, domainId, keyId), "");
+			}
+		} else {
+			success = false;
+			break;
+		}
+	}
+
+	wait(delayJittered(1.0)); // simulate network delay
+
+	success ? req.reply.send(rep) : req.reply.sendError(encrypt_key_not_found());
+
+	return Void();
+}
+
+static Standalone<BlobMetadataDetailsRef> createBlobMetadata(BlobMetadataDomainId domainId) {
+	Standalone<BlobMetadataDetailsRef> metadata;
+	metadata.domainId = domainId;
+	// 0 == no partition, 1 == suffix partitioned, 2 == storage location partitioned
+	int type = deterministicRandom()->randomInt(0, 3);
+	int partitionCount = (type == 0) ? 0 : deterministicRandom()->randomInt(2, 12);
+	fmt::print("SimBlobMetadata ({})\n", domainId);
```

Comment:
a nit: should we be using TraceEvent to log instead? The code runs in simulation so OK to be noisy in logging if needed.

Reply:
Yeah I think if they are useful for debugging, then better to add it as `TraceEvent`, we can choose `SevDebug` to avoid logging in production environments